### PR TITLE
[JIT-1412] Mempool example bugfixes

### PR DESCRIPTION
**Pull request by esemeniuc**
Notes:
- Upgrade Solana version
- Fixes broken mempool example in README
- Perform pubkey verification at CLI startup

Fixes multiple pubkey issue:
```
➜  searcher-examples ✗ cargo run --bin jito-searcher-cli -- \
  --block-engine-url https://dallas.testnet.block-engine.jito.wtf \
  --keypair-path blockengine_testnet.json \
  mempool-accounts --accounts 7heQNXEtxSv3wV8sNbuQsDN5xNGbbpLvHGiyXuJxEf7n 96gYZGLnJYVFmbjzopPSU6QiEV5fGqZNyN9nmNhvrZU5
    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
     Running `target/debug/jito-searcher-cli --block-engine-url 'https://dallas.testnet.block-engine.jito.wtf' --keypair-path /home/eric/dev/block-engine/blockengine_testnet.json mempool-accounts --accounts 7heQNXEtxSv3wV8sNbuQsDN5xNGbbpLvHGiyXuJxEf7n 96gYZGLnJYVFmbjzopPSU6QiEV5fGqZNyN9nmNhvrZU5`
error: unexpected argument '96gYZGLnJYVFmbjzopPSU6QiEV5fGqZNyN9nmNhvrZU5' found

For more information, try '--help'.
```

### CODE REVIEW

File: cli/src/main.rs [hunk 1]
```diff
@@ -48,15 +48,15 @@ struct Args {
 enum Commands {
     /// Subscribe to mempool accounts
     MempoolAccounts {
```
Comment (jedleggett) [inline cli/src/main.rs:50]: these subcommands are confusing, as they read more like params

maybe add verb in name, e.g. `...  SubscribeMempoolAccounts --acounts ...`

File: cli/src/main.rs [hunk 2]
```diff
@@ -188,10 +188,11 @@ async fn main() {
...
         }
         Commands::MempoolAccounts { accounts } => {
-            info!(
-                "waiting for mempool transactions that write-locks accounts: {:?}",
-                accounts
-            );
+            let accounts = accounts
+                .into_iter()
+                .map(|a| a.to_string())
+                .collect::<Vec<String>>();
```
Comment (jedleggett) [inline cli/src/main.rs:194]: why have Pubkey input if just converting to string anyway?  why not just leave as string param?

**Review by jedleggett — APPROVED** [review]
couple nits, but lgtm

File: cli/src/main.rs (hunk 2 above)
Comment (esemeniuc) [reply cli/src/main.rs:194]: Main reason is to have type info earlier, and exit sooner if incorrect params are passed

File: cli/src/main.rs (hunk 1 above)
Comment (esemeniuc) [reply cli/src/main.rs:50]: updated the readme to be more helpful. will defer to @buffalu

**Pull request merged**